api/test/scenarios-pu-data/steps/given-scenario-pu-data-exists.ts

```diff
@@ -0,0 +1,47 @@
+import { INestApplication } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { v4 } from 'uuid';
+
+import { ScenariosPlanningUnitGeoEntity } from '../../../src/modules/scenarios-planning-unit/entities/scenarios-planning-unit.geo.entity';
+import { remoteConnectionName } from '../../../src/modules/scenarios-planning-unit/entities/remote-connection-name';
+import { LockStatus } from '../../../src/modules/scenarios-planning-unit/lock-status.enum';
+
+export const GivenScenarioPuDataExists = async (
+  app: INestApplication,
+  scenarioId = v4(),
+): Promise<{
+  scenarioId: string;
+  projectId: number;
+  rows: ScenariosPlanningUnitGeoEntity[];
+}> => {
+  const repo: Repository<ScenariosPlanningUnitGeoEntity> = await app.get(
+    getRepositoryToken(ScenariosPlanningUnitGeoEntity, remoteConnectionName),
+  );
+  const projectId = 1234;
```

Comment:
Given my other note about the role of the `puid` column, I suggest instead to make this a numeric sequence (and ideally to also delete the parent scenario in the `afterEach()` step of the test above, just for general cleanup as we won't need to  deal with scenario id clashes due to using `v4()` to generate these.

Reply:
There aren't any `FKs` so it is possible to create rows without any relations (and thus, we don't need scenario there at all, nor its removal)

Reply:
you're totally right - I missed that we're only inserting scenarios_pu_data rows here.

still best to clean these rows up (by the v4()-generated scenarioId?) on teardown to avoid keeping test data around if this is run against a volume-backed db (rather than the ephemeral dbs in the CI pipeline)?

Reply:
`id` fixed in https://github.com/Vizzuality/marxan-cloud/pull/148/commits/6396c14d671e06fbc11259fc539499d536454a85

Reply:
`cleanup` fixed in https://github.com/Vizzuality/marxan-cloud/pull/148/commits/e7c84aeaad8622b4c1682b9fd953add6f7395bbb 
Refactored steps a bit.

Reply:
Ok I didn't constrain this PU because we need to also allow User uploads with their own ids. This ids are also generated at insert time so we can control it but a unique constrain based on scenario id should be added. Task for a next pr

Reply:
I have added this to MARXAN-398